RecoLocalTracker/SiPixelRecHits/src/PixelCPEClusterRepair.cc

```diff
@@ -0,0 +1,628 @@
+/// #include "RecoLocalTracker/SiPixelRecHits/interface/SiPixelTemplateSplit.h"
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include <vector>
+#include "boost/multi_array.hpp"
+
+#include <iostream>
+
+using namespace SiPixelTemplateReco;
+//using namespace SiPixelTemplateSplit;
+using namespace std;
+
+namespace {
+   constexpr float micronsToCm = 1.0e-4;
+   constexpr int cluster_matrix_size_x = 13;
+   constexpr int cluster_matrix_size_y = 21;
+}
+
+//-----------------------------------------------------------------------------
+//  Constructor.
+//
+//-----------------------------------------------------------------------------
+PixelCPEClusterRepair::PixelCPEClusterRepair(edm::ParameterSet const & conf,
+                                           const MagneticField * mag,
+                                           const TrackerGeometry& geom,
+                                           const TrackerTopology& ttopo,
+                                           const SiPixelLorentzAngle * lorentzAngle,
+					   const SiPixelTemplateDBObject * templateDBobject,
+					   const SiPixel2DTemplateDBObject * templateDBobject2D )
+: PixelCPEBase(conf, mag, geom, ttopo, lorentzAngle, nullptr, templateDBobject, nullptr,1)
+{
+   //cout << endl;
+   //cout << "Constructing PixelCPEClusterRepair::PixelCPEClusterRepair(...)................................................." << endl;
+   //cout << endl;
+   
+   // Configurable parameters
+   //DoCosmics_ = conf.getParameter<bool>("DoCosmics"); // Not used in templates
+   //LoadTemplatesFromDB_ = conf.getParameter<bool>("LoadTemplatesFromDB"); // Moved to Base
+   
+   //cout << " PixelCPEClusterRepair : (int)LoadTemplatesFromDB_ = " << (int)LoadTemplatesFromDB_ << endl;
+   //cout << "field_magnitude = " << field_magnitude << endl;
+   
+   // configuration parameter to decide between DB or text file template access
+   
+   if ( LoadTemplatesFromDB_ )
+   {
+      //cout << "PixelCPEClusterRepair: Loading templates from database (DB) --------- " << endl;
+      
+      // Initialize template store to the selected ID [Morris, 6/25/08]
+      if ( !SiPixelTemplate::pushfile( *templateDBobject_, thePixelTemp_) )
+         throw cms::Exception("PixelCPEClusterRepair")
+         << "\nERROR: Templates not filled correctly. Check the sqlite file. Using SiPixelTemplateDBObject version "
+         << (*templateDBobject_).version() << "\n\n";
+
+      // Initialize template store to the selected ID [Morris, 6/25/08]
+      if ( !SiPixelTemplate2D::pushfile( *templateDBobject2D , thePixelTemp2D_) )
+         throw cms::Exception("PixelCPEClusterRepair")
+         << "\nERROR: Templates not filled correctly. Check the sqlite file. Using SiPixelTemplateDBObject version "
+         << (*templateDBobject_).version() << "\n\n";
+   }
+   else
+   {
+      //cout << "PixelCPEClusterRepair : Loading templates for barrel and forward from ASCII files ----------" << endl;
```

Comment:
Please remove the commented out lines, or replace with some LogDebug or similar

Reply:
Actually, I'd like a message like this to be visible in production jobs.  What kind of LogXYZ do I need to do that?  That is, I want to log it, and not just as debug.  Is that LogInfo or something?

Reply:
Now the message is commented out.
For the level of logging that better suits your needs please refer to 
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideMessageLogger
In any case, please make sure that the output doesn't get filled with tons of replicas of your messages